### ISSUE TITLE
`pkgs.foobar` has no nativeDrv in nixos modules

### ISSUE THREAD

**Issue by bennofs** [NixOS/nixpkgs#51057]
When using packages from the `pkgs` arguments of NixOS modules, that package won't have any `nativeDrv` or `crossDrv` (it is not spliced). This has the implication that the following won't work inside NixOS modules when cross compiling:

```
  let customizedFoobar = pkgs.foobar.override { hello = config.mymodule.hello-package };
```

If I set `mymodule.hello-package = pkgs.hello` when using the module, that'll break cross compilation since `foobar` will then get a non-spliced hello as argument (thus `nativeBuildInputs` are not working correctly anymore). In some cases it is possible to work around this by directly using `buildPackages` but that seems ugly. Is there a better solution? Should we make `pkgs` be the spliced `pkgs` in NixOS modules at least?

**Comment by bennofs** [comment]
This is a similar issue to #49526

**Comment by bennofs** [comment]
I don't think the relevant code has changed.

**Comment by Ericson2314** [comment]
Perhaps we can combine #49526 as it's the same underlying issue?

**Comment by bennofs** [comment]
Ah, I think that's fair. Closed in favor us  #49526.
